test/e2e/frontend/environments/smtp.js

```diff
@@ -0,0 +1,49 @@
+const childProcess = require('child_process')
+
+const mailpitLogger = (data) => {
+    console.info(`Mailpit: ${data}`)
+}
+
+module.exports = async function (config = {}) {
+    let { webPort, smtpPort } = config
+
+    webPort = webPort || 8025
+    smtpPort = smtpPort || 1025
+
+    mailpitLogger('Starting e-mail server...')
+
+    const dockerCommand = 'docker'
+    const args = [
+        'run',
+        '--rm',
+        `-p=${webPort}:8025`,
+        `-p=${smtpPort}:1025`,
+        '-e=MP_MAX_MESSAGES=5000',
+        '-e=MP_SMTP_AUTH_ACCEPT_ANY=1',
+        '-e=MP_SMTP_AUTH_ALLOW_INSECURE=1',
+        'axllent/mailpit'
+    ]
```

Comment:
Should we be starting this as a child process or using a GHA docker service (and have people start it manually for local testing?) 

See how we use a Postgres container here https://github.com/FlowFuse/flowfuse/blob/0d5e6ec5e778ffb7aace9e34294787df1421177c/.github/workflows/tests.yml#L98

Reply:
That was my initial thought as well, but we need the service to be available when we're testing locally as well without adding additional complexity for developers.

Reply:
OK, so only other question is does this clean up properly when the tests exit? I see the `--rm` so the container will get removed when shutdown, and I assume the child process dies with the test process which triggers an exit?

Reply:
Yes, the process is killed when the parent process is because it's not running in a detached state. The `--rm` is just a precaution